paddle/framework/details/buffered_channel.h

```diff
@@ -67,20 +71,30 @@ bool Buffered<T>::Send(T* item) {
     empty_cond_var_.notify_one();
     ret = true;
   }
+  send_ctr--;
+  destructor_cond_var_.notify_one();
   return ret;
 }
 
 template <typename T>
 bool Buffered<T>::Receive(T* item) {
+  bool ret = false;
+  // Once the channel has been closed and all data has been consumed,
+  // just return false. Don't even try acquiring the mutex.
```

Comment:
If a reader thread is allowed to acquire mutex here, then in a hypothetical infinite readers scenario, the destructor could be blocked forever. So after closing and when all the data in the channel is consumed, any call to `Receive` should immediately return without even trying to acquire the lock.